content/blog/improved-kubernetes-await-logic/index.md

```diff
@@ -0,0 +1,293 @@
+   A Pod's [termination message](https://kubernetes.io/docs/tasks/debug/debug-application/determine-reason-pod-failure/) can be helpful in `CrashLoopBackOff` situations and is read from `/dev/termination-log` by default.
+
+   (You can configure this path with `terminationMessagePath`, or use `terminationMessagePolicy: FallbackToLogsOnError` to use a Pod's logs as its termination message.)
+
+## Faster deletions with "pulumi.com/deletionPropagationPolicy"
+
+Pulumi uses [foreground](https://kubernetes.io/docs/concepts/architecture/garbage-collection/#foreground-deletion) cascading deletion by default, which deletes a resource _and_ all of its dependents.
+This is the most robust form of deletion but it is also the slowest, and until now there was no way to safely customize this behavior.
+
+The new `pulumi.com/deletionPropagationPolicy` annotation allows you to customize delete behavior while still allowing Pulumi to await readiness when creating or updating the resource.
+
+The annotation accepts three propagation policies:
+
+1. "background": delete the owner resource and leave dependent resources to be asynchronously garbage collected.
+   This is faster than "foreground" deletion propagation, but dependent resources can remain temporarily or even indefinitely if they are not finalized.
+2. "orphan": delete the owner resource and leave dependent resources untouched.
+   This can be useful if you want to keep resources around for migration or debugging purposes.
+3. "foreground": the default behavior of deleting the resource and all of its dependents.
+   This is slower but guarantees all dependents have been cleaned up if it succeeds.
+
+{{% notes type="info" %}}
+Using `skipAwait` to speed up deletion is not recommended when server-side apply is enabled because it is not respected by all resources and it can lead to race conditions during replacement.
+The current behavior is considered buggy, and future releases might change the behavior of the `skipAwait` annotation to have no effect during deletion.
+
+The `pulumi.com/deletionPropagationPolicy` annotation is the recommended way to delete something quickly and safely.
+{{% /notes %}}
+
+## Experimental: Custom readiness with "pulumi.com/waitFor"
+
+{{% notes type="info" %}}
+The `pulumi.com/waitFor` annotation is considered experimental and its syntax may change in the future.
+{{% /notes %}}
+
+There is no one-size-fits-all definition for what it means for a Kubernetes resource to be "ready," so we have also introduced a new annotation, `pulumi.com/waitFor`, which allows you to specify custom readiness criteria on a per-resource basis.
+This is functionally similar to `kubectl` [wait](https://kubernetes.io/docs/reference/kubectl/generated/kubectl_wait/).
+
+The new `pulumi.com/waitFor` annotation can be applied to any non-[Patch](https://www.pulumi.com/blog/kubernetes-server-side-apply/) resource with metadata, and it accepts three possible formats:
```

Comment:
It strikes me as a bug that a patch resource doesn't wait for readiness. Assumedly the patch is causing a spec change, meaning a rollout may occur.  Seems no different from awaiting an update. For purposes of the blog post, could we offer a note on the topic?

Reply:
I actually can't remember why I wrote this, because I don't think this is actually true -- patch resources should be fine.